lib/ansible/plugins/lookup/template.py

```diff
@@ -76,19 +76,18 @@ def run(self, terms, variables, **kwargs):
                 with open(to_bytes(lookupfile, errors='surrogate_or_strict'), 'rb') as f:
                     template_data = to_text(f.read(), errors='surrogate_or_strict')
 
-                    # set jinja2 internal search path for includes
-                    searchpath = variables.get('ansible_search_path')
-                    if searchpath:
-                        # our search paths aren't actually the proper ones for jinja includes.
-                        # We want to search into the 'templates' subdir of each search path in
-                        # addition to our original search paths.
-                        newsearchpath = []
-                        for p in searchpath:
-                            newsearchpath.append(os.path.join(p, 'templates'))
-                            newsearchpath.append(p)
-                        searchpath = newsearchpath
-                    else:
-                        searchpath = [self._loader._basedir, os.path.dirname(lookupfile)]
+                    # set Jinja2 internal search path for includes
+                    searchpath = variables.get('ansible_search_path', [])
+                    searchpath.extend([self._loader.get_basedir(), os.path.dirname(lookupfile)])
```

Comment:
get_basedir should not be needed, the ansible_search_path should include the correct one

Reply:
the lookupfile should not presume a templates/ subdir so we may want to add it always after appending it, not before

Reply:
Thanks, fixed.